File: app/src/main/kotlin/com/alapshin/boilerplate/log/LogUtil.java
package com.alapshin.boilerplate.log;

import timber.log.Timber;

/**
 * Wrapper around {@link Timber}
 *
 * In debug builds logs are written to Logcat.
 * In release builds logs are written to Crashlytics.
 */
public class LogUtil {
    public static void v(String msg) {
        Timber.v(msg);
    }

    public static void v(Throwable tr) {
        Timber.v(tr);
    }

    public static void v(String msg, Throwable tr) {
        Timber.v(tr, msg);
    }

    public static void d(String msg) {
        Timber.d(msg);
    }

    public static void d(Throwable tr) {
        Timber.d(tr);
    }

    public static void d(String msg, Throwable tr) {
        Timber.d(tr, msg);
    }

    public static void i(String msg) {
        Timber.i(msg);
    }

    public static void i(Throwable tr) {
        Timber.i(tr);
    }

    public static void i(String msg, Throwable tr) {
        Timber.i(tr, msg);
    }

    public static void w(String msg) {
        Timber.w(msg);
    }

    public static void w(Throwable tr) {
        Timber.w(tr);
    }

    public static void w(String msg, Throwable tr) {
        Timber.w(tr, msg);
    }

    public static void e(String msg) {
        Timber.e(msg);
    }

    public static void e(Throwable tr) {
        Timber.e(tr);
    }

    public static void e(String msg, Throwable tr) {
        Timber.e(tr, msg);
    }
}

File: app/src/main/kotlin/com/alapshin/boilerplate/log/LogUtil.kt
package com.alapshin.boilerplate.log

import timber.log.Timber

/**
 * Wrapper around [Timber]
 */
object LogUtil {
    fun v(msg: String) {
        Timber.v(msg)
    }

    fun v(tr: Throwable) {
        Timber.v(tr)
    }

    fun v(msg: String, tr: Throwable) {
        Timber.v(tr, msg)
    }

    fun d(msg: String) {
        Timber.d(msg)
    }

    fun d(tr: Throwable) {
        Timber.d(tr)
    }

    fun d(msg: String, tr: Throwable) {
        Timber.d(tr, msg)
    }

    fun i(msg: String) {
        Timber.i(msg)
    }

    fun i(tr: Throwable) {
        Timber.i(tr)
    }

    fun i(msg: String, tr: Throwable) {
        Timber.i(tr, msg)
    }

    fun w(msg: String) {
        Timber.w(msg)
    }

    fun w(tr: Throwable) {
        Timber.w(tr)
    }

    fun w(msg: String, tr: Throwable) {
        Timber.w(tr, msg)
    }

    fun e(msg: String) {
        Timber.e(msg)
    }

    fun e(tr: Throwable) {
        Timber.e(tr)
    }

    fun e(msg: String, tr: Throwable) {
        Timber.e(tr, msg)
    }
}
